src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/SplitIndexWriter.java

```diff
@@ -173,7 +156,9 @@ protected void addLinksForIndexes(Content contentTree) {
             contentTree.add(links.createLink(pathToRoot.resolve(DocPaths.ALLPACKAGES_INDEX),
                                              contents.allPackagesLabel));
         }
-        if (searchItems.containsAnyOfCategories(Category.SYSTEM_PROPERTY)) {
+        boolean anySystemProperties = mainIndex.getItems(Category.TAGS).stream()
+                .anyMatch(sii -> sii.getDocTree().getKind() == DocTree.Kind.SYSTEM_PROPERTY);
+        if (anySystemProperties) {
             contentTree.add(getVerticalSeparator());
             contentTree.add(links.createLink(pathToRoot.resolve(DocPaths.SYSTEM_PROPERTIES),
                                              contents.systemPropertiesLabel));
```

Comment:
1. the getItems code could be simplified by providing `getItems(DocTree.Kind)`
2. This code is common to both `SingleIndexWriter` and `SplitIndexWriter` and so could be moved up into `AbstractIndexWriter`